File: src/main/GO/leetcode/lists/offerII/Offer47.go
package offerII

/**
 * @author wujingxinit@outlook.com
 * @date 2023/1/16 17:52
 */

//后序遍历
func pruneTree(root *TreeNode) *TreeNode {
	if root == nil {
		return nil
	}
	root.Left = pruneTree(root.Left)
	root.Right = pruneTree(root.Right)
	if root.Left == nil && root.Right == nil && root.Val == 0 {
		return nil
	}
	return root
}

File: java/src/leetcode/lists/offerII/Offer47.java
package leetcode.lists.offerII;

import common.TreeNode;

/**
 * @author wujingxinit@outlook.com
 * @date 2023/7/25 11:26
 */
public class Offer47 {

    public TreeNode pruneTree(TreeNode root) {
        if (root == null) {
            return null;
        }
        root.left = pruneTree(root.left);
        root.right = pruneTree(root.right);
        if (root.left == null && root.right == null && root.val == 0) {
            return null;
        }
        return root;
    }
}
